File: backend/src/main/java/com/washer/backend/entity/Device.java
package com.washer.backend.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import lombok.Data;

@Data
@TableName("device")
public class Device {

    @TableId
    private Long id;

    private String deviceCode;
    private Long storeId;
    private Long bayId;
    private String deviceType;
    private String deviceRole;
    private String bayDeviceType;
    private String deviceName;
    private String deviceStatus;
    private String protocolType;
    private String firmwareVersion;
    private LocalDateTime lastHeartbeatTime;
    private LocalDateTime lastOnlineTime;
    private LocalDateTime installTime;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: admin-web/src/types/device.ts
export interface DeviceItem {
  id: number;
  deviceCode?: string;
  deviceName?: string;
  storeId: number;
  storeName?: string;
  deviceType?: string;
  deviceRole?: string;
  status?: string;
  deviceStatus?: string;
  protocolType?: string;
  firmwareVersion?: string;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface DeviceFormPayload {
  storeId: number;
  deviceCode?: string;
  deviceName?: string;
  deviceType?: string;
  deviceRole?: string;
  deviceStatus?: string;
  protocolType?: string;
  firmwareVersion?: string;
  remark?: string;
}
